plugins/module_utils/openshift_adm_prune_auth.py

```diff
@@ -0,0 +1,340 @@
+                        self.fail_json(msg=msg)
+        return candidates, changed
+
+    def auth_prune_roles(self):
+        params = {'kind': 'Role', 'api_version': 'rbac.authorization.k8s.io/v1', 'namespace': self.params.get('namespace')}
+        for attr in ('name', 'label_selectors'):
+            if self.params.get(attr):
+                params[attr] = self.params.get(attr)
+
+        result = self.kubernetes_facts(**params)
+        if not result['api_found']:
+            self.fail_json(msg=result['msg'])
+
+        roles = result.get('resources')
+        if len(roles) == 0:
+            self.exit_json(changed=False, msg="No candidate rolebinding to prune from namespace %s." % self.params.get('namespace'))
+
+        ref_roles = [(x['metadata']['namespace'], x['metadata']['name']) for x in roles]
+        candidates = self.prune_resource_binding(kind="RoleBinding",
+                                                 api_version="rbac.authorization.k8s.io/v1",
+                                                 ref_kind="Role",
+                                                 ref_namespace_names=ref_roles,
+                                                 propagation_policy='Foreground')
+        if len(candidates) == 0 or self.check_mode:
+            self.exit_json(changed=False, role_binding=candidates)
+
+        self.exit_json(changed=True, role_binding=candidates)
+
+    def auth_prune_clusterroles(self):
+        params = {'kind': 'ClusterRole', 'api_version': 'rbac.authorization.k8s.io/v1'}
+        for attr in ('name', 'label_selectors'):
+            if self.params.get(attr):
+                params[attr] = self.params.get(attr)
+
+        result = self.kubernetes_facts(**params)
+        if not result['api_found']:
+            self.fail_json(msg=result['msg'])
+
+        clusterroles = result.get('resources')
+        if len(clusterroles) == 0:
+            self.exit_json(changed=False, msg="No clusterroles found matching input criteria.")
+
+        ref_clusterroles = [(None, x['metadata']['name']) for x in clusterroles]
+
+        # Prune ClusterRoleBinding
+        candidates_cluster_binding = self.prune_resource_binding(kind="ClusterRoleBinding",
+                                                                 api_version="rbac.authorization.k8s.io/v1",
+                                                                 ref_kind=None,
+                                                                 ref_namespace_names=ref_clusterroles)
+
+        # Prune Role Binding
+        candidates_namespaced_binding = self.prune_resource_binding(kind="RoleBinding",
+                                                                    api_version="rbac.authorization.k8s.io/v1",
+                                                                    ref_kind='ClusterRole',
+                                                                    ref_namespace_names=ref_clusterroles)
+        if self.check_mode:
```

Comment:
It's unclear to me how `check_mode` is used here. If there would be a change, `check_mode` must report that change (changed=True), but without really performing it (only simulating it). This is the logic behind `check_mode` inside the AWS collection for example. So, maybe I missed something here and I am wrong about, I don't know.

Reply:
you are right I just missed it